include/pressio/solvers_nonlinear/impl/updaters/solvers_basic_backtrack.hpp

```diff
@@ -0,0 +1,128 @@
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+// contributors may be used to endorse or promote products derived
+// from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+// FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+// COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+// INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+// HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
+// IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Francesco Rizzi (fnrizzi@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#ifndef SOLVERS_NONLINEAR_IMPL_UPDATERS_SOLVERS_BASIC_BACKTRACK_HPP_
+#define SOLVERS_NONLINEAR_IMPL_UPDATERS_SOLVERS_BASIC_BACKTRACK_HPP_
+
+namespace pressio{ namespace nonlinearsolvers{ namespace impl{
+
+template <typename StateType>
+class BasicBacktrackUpdater
+{
+  StateType trialState_;
+
+public:
+  BasicBacktrackUpdater() = delete;
+  BasicBacktrackUpdater(BasicBacktrackUpdater const &) = default;
+  BasicBacktrackUpdater & operator=(BasicBacktrackUpdater const &) = default;
+  BasicBacktrackUpdater(BasicBacktrackUpdater &&) = default;
+  BasicBacktrackUpdater & operator=(BasicBacktrackUpdater &&) = default;
+  ~BasicBacktrackUpdater() = default;
+
+  BasicBacktrackUpdater(const StateType & state)
+    : trialState_(::pressio::ops::clone(state))
+  {
+    setTrialStateToZero();
+  }
+
+public:
+  void reset(){ /* no op */ }
+
+  template<typename system_t, typename solver_mixin_t>
+  void operator()(const system_t & system,
+		  StateType & state,
+		  solver_mixin_t & solver)
+  {
+    PRESSIOLOG_DEBUG("BasicBacktrack update");
+    setTrialStateToZero();
+    const auto & p_k   = solver.correctionCRef();
+    auto fx_k    = solver.residualNormCurrentCorrectionStep();
+
+    using scalar_type = decltype(fx_k);
+    constexpr auto one = ::pressio::utils::Constants<scalar_type>::one();
+    auto alpha = static_cast<scalar_type>(1);
+    fx_k = std::pow(fx_k, ::pressio::utils::Constants<scalar_type>::two());
+
+    PRESSIOLOG_DEBUG("starting backtracking");
+    scalar_type ftrial = {};
+    bool done = false;
+    while (not done)
+    {
+      if (std::abs(alpha) <= 0.001){
```

Comment:
can you please write a comment here to summarize the choice for 0.001 ? basically what we said in slack, so that we don't forget 1 year from now. ifyou have any reference to what libraries use great, otherwise the "adimensional" argument that you said to me is good to have

Reply:
Added